### PR TITLE
Update pre-commit Hook Versions

### PR DESCRIPTION
**Pull request by mcdonnnj**
# 

## 🗣 Description
In an effort to keep things up to date I am trying to run a `pre-commit autoupdate` monthly to refresh hooks. This is the result of this effort.

The [pre-commit/mirrors-isort](https://github.com/pre-commit/mirrors-isort) repository has been deprecated since `isort` now contains appropriate code to run as a `pre-commit` hook. The repository used in the `pre-commit` configuration have been updated appropriately.

We have seen a number of issues from using the `terraform_validate` hook.  These have primarily been related to proxy providers and Terraform code using remote state. The issue with remote state can be read about at https://github.com/hashicorp/terraform/issues/21408, and the issue with proxy providers has been wrapped into that issue as described in this reply: https://github.com/hashicorp/terraform/issues/21706#issuecomment-501497300. There is an accepted PR (https://github.com/hashicorp/terraform/pull/24887) for remote state that is included in the future Terraform 0.13 release. I am unsure if it will resolve issues with proxy providers as there is another PR (https://github.com/hashicorp/terraform/pull/24896) waiting for feedback that attempts to deal with that specifically.

## 💭 Motivation and Context
Updates are good, fixing problems is good, things are good 👍

## 🧪 Testing
All automated tests pass.

## 🚥 Types of Changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (causes existing functionality to change)

## ✅ Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -61,31 +61,41 @@ repos:
...
+      # We have seen issues primarily with proxy providers and Terraform code
+      # that uses remote state. The PR
+      # https://github.com/hashicorp/terraform/pull/24887
+      # has been approved and is part of the 0.13 release to resolve the issue
+      # with remote states.
+      # The PR
+      # https://github.com/hashicorp/terraform/pull/24896
+      # is a proprosed fix to deal with `terraform validate` with proxy
+      # providers (among other configurations).
+      # - id: terraform_validate
```
Comment (dav3r) [inline .pre-commit-config.yaml:95]: This comment is very thorough, but I think it needs one final sentence to the effect of "We have decided to disable the terraform_validate hook until the the issues above have been resolved, which we hope will be when Terraform 0.13 is released."

**Review by dav3r — APPROVED** [review]
Approved, though please see my request for an addition to a comment.
↪️

**Review by jsf9k — APPROVED** [review]
Strong work.  Approved, but please see @dav3r's comment.

**Review by felddy — APPROVED** [review]
Wunderbar!

File: .pre-commit-config.yaml (hunk 1 above)
Comment (mcdonnnj) [reply .pre-commit-config.yaml:95]: Good point. I've added a sentence to that effect in f831ef8.

**Review by hillaryj — APPROVED** [review]
🎉

**Comment by jsf9k** [comment]
:shipit:

**Pull request merged**